staging/src/k8s.io/client-go/openapi/groupversion.go

```diff
@@ -18,15 +18,24 @@ package openapi
 
 import (
 	"context"
+	"encoding/json"
 
 	openapi_v3 "github.com/google/gnostic/openapiv3"
 	"google.golang.org/protobuf/proto"
 	"k8s.io/kube-openapi/pkg/handler3"
+	"k8s.io/kube-openapi/pkg/spec3"
 )
 
 const openAPIV3mimePb = "application/com.github.proto-openapi.spec.v3@v1.0+protobuf"
+const jsonMime = "application/json"
 
 type GroupVersion interface {
+	// Raw data types
+	SchemaJSON() ([]byte, error)
+	SchemaPB() ([]byte, error)
+
+	// Serialized types
+	SchemaOpenAPI() (*spec3.OpenAPI, error)
 	Schema() (*openapi_v3.Document, error)
```

Comment:
Why is it the responsibility of this interface to couple SchemaJSON, Schema, SchemaOpenAPI, SchemaPB? I would just decouple the source and the serialization (or the other way around, I'm not even sure).

Reply:
I would prefer to remove `Schema()`, and just add `SchemaJSON()` and `SchemaPB()`?

Reply:
Go for it, I think it's probably better.